hw/ip/spi_host/data/spi_host.hjson

```diff
@@ -0,0 +1,590 @@
+          resval: "0x0"
+        },
+        { bits: "30",
+          name: "RST_FSM",
+          desc: '''This bit holds the control FSM in reset when set to 1.'''
+          resval: "0x0",
+          hwqe: "true",
+          hwacces: "hrw"
+        },
+        { bits: "29",
+          name: "RST_TXFIFO",
+          desc: '''Resets the TX FIFO when set to 1.'''
+          resval: "0x0",
+        },
+        { bits: "28",
+          name: "RST_RXFIFO",
+          desc: '''Resets pulse the RX FIFO when set to 1.'''
+          resval: "0x0"
+        },
+        { bits: "27",
+          name: "PASSTHRU",
+          desc: '''Enables Pass-through mode, wherein the SPI_HOST IP surrenders
+                   control of the SPI bus to another block. In this mode the
+                   `sck`, `cs_n` and `sd` signals are no longer driven by
+                   the SPI_HOST IP, rather they are multiplexed out to match the
+                   on-chip inputs `pt_sck_i`, `pt_cs_ni`, `pt_sdt_i`, `pt_sdo_i`,
+                   and `pt_sdi_o`.  (Since the `sd` bus is bidirectional, there are
+                   separate input, output and tri-state controls for this bus.)''',
+          resval: "0x0"
+        },
+        { bits: "24:16",
+          name: "TX_WATERMARK"
+          desc: '''The IP can optionally send an interrupt if the
+                   depth of the TX FIFO is ever less than
+                   TX_WATERMARK words (32b each). Set to zero
+                   to temporarily disable interrupts.'''
+          resval: "0"
+        },
+        { bits: "15:9",
+          name: "RX_WATERMARK"
+          desc: '''The IP can optionally send an interrupt if the
+                   depth of the RX FIFO is ever greater than
+                   RX_WATERMARK words (32b each).'''
+          resval: "127"
+        },
+        { bits: "8",
+          name: "MANCS_EN",
+          desc: '''Enable Manual Chip Select. Typically each command
+                   manages the CS lines automatically, issuing a command
+                   to exactly one device, and lowering and raising the
+                   appropriate CS line after transmitting or receiving the
+                   necessary bytes.  Asserting this bit overrides this
+                   automatic selection, instead setting the CS lines to
+                   reflect the contents of the MANUAL_CS field.'''
+          resval: "0x0"
+        },
+        { bits: "7:0",
+          name: "MANUAL_CS"
+          desc: '''When MANCS_EN = 1, the cs_n lines are directly to the
+                   value of MANUAL_CS.'''
+          resval: 0
+        }
+      ]
+    },
+    { name: "STATUS",
+      desc: "Status register",
+      swaccess: "ro",
+      hwaccess: "hwo",
+      fields: [
+        { bits: "31",
+          name: "READY",
+          desc: '''When high, indicates the SPI host is ready to receive
```

Comment:
there's a note in the spec about being able to write another command even if active is high. Does this mean this ready clears much sooner than active potentially? (i guess after the command gets sync'd?)

Reply:
That's right.